src/ImageSharp.Web/ImageSharpTagHelper.cs

```diff
@@ -0,0 +1,365 @@
+            }
+
+            if (this.ResizeMode.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Mode, this.ResizeMode.Value.ToString());
+            }
+
+            if (this.AnchorPosition.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Anchor, this.AnchorPosition.Value.ToString());
+            }
+
+            if (this.Center.HasValue)
+            {
+                string xy = $"{this.Center.Value.X.ToString(this.parserCulture)}{this.separator}{this.Center.Value.Y.ToString(this.parserCulture)}";
+                commands.Add(ResizeWebProcessor.Xy, xy);
+            }
+
+            if (this.PadColor.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Color, this.PadColor.Value.ToHex());
+            }
+
+            if (this.Compand.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Compand, this.Compand.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Orient.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Orient, this.Orient.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Sampler.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Sampler, this.Sampler.Value.Name);
+            }
+        }
+
+        private void AddAutoOrientCommands(CommandCollection commands)
+        {
+            if (this.AutoOrient.HasValue)
+            {
+                commands.Add(AutoOrientWebProcessor.AutoOrient, this.AutoOrient.Value.ToString());
+            }
+        }
+
+        private void AddFormatCommands(CommandCollection commands)
+        {
+            if (this.Format.HasValue)
+            {
+                commands.Add(FormatWebProcessor.Format, this.Format.Value.Name);
+            }
+        }
+
+        private void AddBackgroundColorCommands(CommandCollection commands)
+        {
+            if (this.BackgroundColor.HasValue)
+            {
+                commands.Add(BackgroundColorWebProcessor.Color, this.BackgroundColor.Value.ToHex());
+            }
+        }
+
+        private void AddQualityCommands(CommandCollection commands)
+        {
+            if (this.Quality.HasValue)
+            {
+                commands.Add(QualityWebProcessor.Quality, this.Quality.Value.ToString(this.parserCulture));
+            }
+        }
+
+        private static string AddQueryString(
+            string uri,
+            CommandCollection commands)
+        {
+            int anchorIndex = uri.IndexOf('#');
+            string uriToBeAppended = uri;
+            string anchorText = string.Empty;
+
+            // If there is an anchor, then the query string must be inserted before its first occurrence.
+            if (anchorIndex != -1)
+            {
+                anchorText = uri.Substring(anchorIndex);
+                uriToBeAppended = uri.Substring(0, anchorIndex);
```

Comment:
span? could look at QueryHelpers.AddQueryString for the span variant

Reply:
Good catch. Much better now!